proposals/lip-Introduce_cross_chain_messages.md

```diff
@@ -0,0 +1,650 @@
+The [token ID][token-LIP-tokenID] for this fee is always the token ID of the LSK token and as such is not repeated in the message. 
+The LSK token is the main utility token of the Lisk ecosystem and as such is the only good candidate for paying cross-chain fees.
+
+
+#### Status
+
+The basic error handling for routing messages to other chains is done by the mainchain. 
+For example, in the case the recipient chain does not exist, is not active or has been terminated, the mainchain will return the message to the sending chain. 
+The sending chain can then revert the message and potentially refund users. 
+This design choice allows sidechains to send messages to other chains without needing to monitor the status (or even existence) of every other chain. 
+Information about the reason why the message failed and the initial message identifier are stored in the `status` property.
+
+The constant table lists the different status codes defined by the Interoperability module. 
+To allow for future updates or improvements of the Interoperability module, we restrict other modules from using status codes up to 63 (included). 
+Other modules may use other status codes larger or equal to 64.
+
+
+#### Parameters
+
+The `params` property of the messages is defined by each module and can follow any schema, similar to the `params` property of a transaction. 
+The `params` property is not deserialized or validated by the Interoperability module.
+
+In the Lisk ecosystem, all cross-chain messages are routed through the mainchain. 
+This means that messages should always have a sufficiently small size in order to be easily included in mainchain blocks. 
+As the mainchain payload size limit is 15 KiB, and [other properties][CCU-LIP] in the cross-chain updates will not be larger than 4 KiB, we limit the message size to 10 KiB. 
+To guarantee that all messages can be included and handled, sidechains in the Lisk ecosystem should have a payload size limit equal to or greater than 15 KiB (15 x 1024 bytes).
+
+
+### Message ID
+
+As all messages emitted in the Lisk ecoystem are distinct (thanks to the `index` property), we can assign a unique ID to them.
+The message ID is obtained in the same way as other IDs in the protocol, namely by hashing the serialized object.
+This assigns to each message a 32-byte value that can be used whenever the message needs to be referenced.
+
+
+### Message Tracking
+
+Tracking messages throughout the ecosystem could be a challenge, particularly if the messages are errored (and hence spawn new messages from the erroring chain).
+However, this can easily be done using the logs emitted by the CCM processing. 
+Indeed, messages which are errored and which send a response message to their sending chain, will log the ID of the return message,
+this allows UI products to display the current status of CCMs (or the status of the returned message if needed).
```

Comment:
```suggestion
allowing off-chain services to display the current status of CCMs (or the status of the returned message if needed).
```

Reply:
To reuse the language we introduce in the module LIPs (and probably in the chain architecture LIP)

Reply:
done